app/src/main/java/com/example/neighborly/RequestActivity.java

```diff
@@ -22,36 +21,47 @@
 import com.google.firebase.database.DatabaseReference;
 import com.google.firebase.database.FirebaseDatabase;
 
-import de.hdodenhof.circleimageview.CircleImageView;
-
 public class RequestActivity extends AppCompatActivity {
+
+    static public final String REQUEST_PRIVATE_CHAT = "private chat";
+    static public final String REQUEST_ITEM = "request item";
+
     private FirebaseRecyclerAdapter<MessageModel, MessageAdapter.MessageHolder> adapter;
     private FirebaseDatabase database;
-    private TextView textViewItemName;
+    private TextView privateChatTitle;
     private EditText input;
     private UserModel curUser;
+    private BuildingModel curBuilding;
+    private String msgPath;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_request);
 
-        Intent intent = getIntent();
-        String itemId = intent.getStringExtra("itemId");
-        String itemName = intent.getStringExtra("itemName");
         curUser = UserModelDataHolder.getInstance().getCurrentUser();
+        curBuilding = BuildingModelDataHolder.getInstance().getCurrentBuilding();
+
+        Intent intent = getIntent();
+        String requestType = intent.getStringExtra("requestType");
+
+        if (requestType.equals(RequestActivity.REQUEST_PRIVATE_CHAT)){
+            String otherUser = intent.getStringExtra("neighbor");
+            handlePrivateChat(otherUser);
+        }
+        if (requestType.equals(RequestActivity.REQUEST_ITEM)){
+            // (Delete comment when the intent is actually sent)
+            // This is supposed to be the requestId as hold in the buildingModel, there  is stored all the data needed.
+            String requestId = intent.getStringExtra("requestId");
+            handleItemRequest(requestId);
+        }
 
-        input = findViewById(R.id.editTextInput);
-        textViewItemName = findViewById(R.id.textViewItemName);
         final RecyclerView recyclerView = findViewById(R.id.recycleView);
         final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
         linearLayoutManager.setStackFromEnd(true);
         recyclerView.setLayoutManager(linearLayoutManager);
-        textViewItemName.setText(itemName);
 
-        String path = "messages/" + itemId;
         database = FirebaseDatabase.getInstance();
-        final DatabaseReference messagesRef = database.getReference(path);
+        final DatabaseReference messagesRef = database.getReference(msgPath);
```

Comment:
I see msgPath gets a value only in the handlePrivateChat scenario and not in handleItemRequest, is it not needed there? or is it just a part that is not implemented yet?

Reply:
You are right. Its implemented in the next pull request. I had to decide how to store everything in the database, and I did it first to the private conversation.
Spoiler, the public requests will be stored under the messages with their request ID. :)

Reply:
![Alt Text](https://media1.tenor.com/images/fc5162c340f4fac6a10991716d7b0bb0/tenor.gif?itemid=11515428)